src/Polysemy/Final.hs

```diff
@@ -0,0 +1,170 @@
+{-# LANGUAGE TemplateHaskell #-}
+module Polysemy.Final
```

Comment:
How much of this module should be exposed in `Polysemy`?

Reply:
My thinking is that I expose only the datatype, `runFinal`, `embedToFinal`, and `embedFinal`, together with adding a tip about the `Polysemy.Final` module for more advanced uses of `Final`.